src/main/java/org/candlepin/service/CloudRegistrationAdapter.java

```diff
@@ -44,6 +45,24 @@ public interface CloudRegistrationAdapter {
     String resolveCloudRegistrationData(CloudRegistrationInfo cloudRegInfo)
         throws CloudRegistrationAuthorizationException, MalformedCloudRegistrationException;
 
+    /**
+     * Resolves the cloud registration details to a specific owner using version 2 logic.
+     *
+     * @param cloudRegInfo
+     *     A CloudRegistrationInfo instance which contains the cloud provider details to process
+     *
+     * @throws MalformedCloudRegistrationException
+     *     if the cloud registration info is null, incomplete, or invalid
+     *
+     * @throws CloudRegistrationAuthorizationException
+     *     if cloud registration is not permitted for the provider or account holder
+     *
+     * @return the cloud authentication result which contains the owner ID, cloud account ID, and
```

Comment:
Something very critical to note here: we will be getting back an owner *key*, not an ID. The owner ID is internal to CP and nothing outside of us uses it. Other teams in the flow may call the value an owner ID, but anywhere we have an external reference to an owner, it's the key.

Reply:
Good catch! Updated to owner key